packages/react/src/DataTable/DataTable.tsx

```diff
@@ -50,6 +50,13 @@ export type DataTableProps<Data extends UniqueRow> = {
    * currently sorted column
    */
   initialSortDirection?: Exclude<SortDirection, 'NONE'>
+
+  /**
+   * Fires every time the user clicks a sortable column header. It reports
+   * the column id that is now sorted and the direction after the toggle
+   * (never `"NONE"`).
+   */
+  onToggleSort?: (columnId: ObjectPaths<Data> | string | number, direction: Exclude<SortDirection, 'NONE'>) => void
```

Comment:
can we add this new prop to the docs file? 🙏  https://github.com/primer/react/blob/main/packages/react/src/DataTable/DataTable.docs.json

Reply:
Done!

Reply:
Had to resolve a merge conflict, too, and opted to force-push a single commit.